js/text-field-character-counter.js

```diff
@@ -0,0 +1,115 @@
+    const initialCharacterCount = el.value ? characterCount(el.value) : 0;
+    // PREVENT COUNTER TO BE ADDED DOUBLE TO TEXTAREAS FOR SOME WEIRD REASON
+    if (!jQuery(el).next().hasClass('composite-character-counters')) {
+      jQuery('<div class="composite-character-counters" style="text-align: right;">Initial Characters: <span class="composite-initial-character-counter" style="margin-right: 5px;">' + initialCharacterCount + '</span>Characters: <span class="composite-character-counter">' + initialCharacterCount + '</span></div>').insertAfter(el);
+    }
+    el.addEventListener('keyup', countCharacters);
+    // TODO Attach eventListener for keyup on the specfic input on focus and remove on blur to save resources
+    // el.addEventListener('focus', addKeyUpListener)
+    // el.addEventListener('blur', addKeyUpListener)
+  }
+
+  function initCounters() {
+    windowReady = true;
+    const widgets = jQuery('.acf-field-flexible-content .layout:not(.acf-clone)');
+    widgets.each(function(index, widget) {
+      const type = jQuery(widget).data('layout');
+      if (!excludedWidgetType(type)) {
+        const textInputs = jQuery(widget).find(getSelectors());
+        textInputs.each(function(index, el) {
+          addCountersAndEventListeners(el);
+        });
+      }
+    })
+    // ADDING TITLE AND DESCRIPTION SEPARATELY
+    addCountersAndEventListeners(jQuery('#title').get(0));
+    addCountersAndEventListeners(jQuery('#acf-field_58abfebd21b82').get(0));
+  }
+
+  let windowReady = false;
+  let isRunning = false;
+  function sumUpAllNonMarkdownFields() {
+    if (isRunning || !windowReady) {
+      return;
+    }
+    isRunning = true;
+    const compositeCounters = jQuery('.composite-character-counter');
+    let total = 0;
+    compositeCounters.each(function(index, item) {
+      const value = item.innerHTML;
+      total += parseInt(value);
+    })
+    document.getElementById('wp-admin-bar-character-count').innerHTML = "<span style='margin: 0 10px;'>Characters: " + total + "</span>";
+    isRunning = false;
+  }
+  
+  function countCharacters(e) {
+    let countTimeout;
+    let el = e.target;
+    jQuery(el).next().find('.composite-character-counter').get(0).innerHTML = el.value.length;
+
+    window.clearTimeout(countTimeout);
+    countTimeout = window.setTimeout(() => {
+      sumUpAllNonMarkdownFields();
+    }, 2000);
+  }
+
+  acf.addAction('append', function($el) {
+    if (!excludedWidgetType($el.data('layout'))) {
+      const textInputs = $el.find(getSelectorsWhenAddingContent());
+      textInputs.each(function(index, el) {
+        addCountersAndEventListeners(el);
+      })
+    }
+  });
+
+  function characterCount(string) {
+    if (string.length === undefined) {
+        return 0;
+    }
+    return string.replace(/\[.*]\(.*\)/g, '')
+      .replace( /^[#]+[ ]+(.*)$/gm, '$1')
+      .replace( /\*\*?(.*?)\*?\*/gm, '$1')
+      .length;
+  }
+
+  function excludedWidgetType(type) {
+    excludedWidgetTypes = [
```

Comment:
Måske lav en whitelist i stedet for en backlist så hvis vi add'er nye widgets kommer de kke bare automatisk med medmindre de explicit bliver added, giver det mening?

Reply:
God ide! Ville bare gerne have haft liste af widgets fra Mikkel Skovbo, inden jeg lavede det... Men jeg fixer det lige efter frokost. 👍🏻